numpy/lib/arraysetops.py

```diff
@@ -519,6 +519,45 @@ def in1d(ar1, ar2, assume_unique=False, invert=False):
     ar1 = np.asarray(ar1).ravel()
     ar2 = np.asarray(ar2).ravel()
 
+    # Check if we can use a fast integer algorithm:
+    integer_arrays = (np.issubdtype(ar1.dtype, np.integer) &
```

Comment:
Thanks, I'll switch to `and`.

I am not sure the boolean types will work - they would require a conversion to uint8 (should I do this?) since using booleans as an index would be interpreted as a mask.

Reply:
No, you would just write `condition1 and condition2` instead of `condition1 & condition2`. Another benefit of this is short-circuiting, which I forgot earlier.

Reply:
> I am not sure the boolean types will work - they would require a conversion to uint8 (should I do this?) since using booleans as an index would be interpreted as a mask.

You could probably use `arr.view(np.uint8)` for a zero-copy view, along with adding a check for `dtype == np.bool_` for each of the arrays. I guess you could just do `if arr.dtype == np.bool_: arr = arr.view(np.uint8)`.

Reply:
Not really. :-) It just creates a "view" into the existing data. https://docs.scipy.org/doc/numpy/reference/generated/numpy.ndarray.view.html

Reply:
> (I suppose it wouldn't really make sense to run np.isin on a boolean/integer ar1/ar2, respectively?)

Try mirroring the existing behaviour, check what that is by running it once.

Reply:
Oh sorry what I mean is: I think both ar1.dtype and ar2.dtype should be bool before applying this np.uint8 conversion. 

If not, one could get an error when running np.isin([3, 1], [False, True, False]). The correct return I think should be [False, False]. However, if we only check one array at a time for boolean and convert it to uint8,  the second array would be converted to unit8 while the first is kept the same, and then [False, True] would be returned.

So requiring both dtypes to be np.bool_ fixes this issue.

Reply:
Adding this now.